src/p2p/potato_socket.erl

```diff
@@ -0,0 +1,56 @@
+-module(potato_socket).
+
+-export([start/0, start_server/0, start_client/0, handle_server_side/1, handle_client_side/1]).
+
+-include_lib("stdlib/include/assert.hrl").
+
+start() ->
+    io:fwrite("hello, world\n"),
+    spawn(?MODULE, start_server, []),
+    timer:sleep(100), % need to sleep otherwise there is a race cond between server/client
```

Comment:
you could wait for a message *started* from the server

Reply:
D: